test/integration/Validator/SymfonyValidatorTest.php

```diff
@@ -0,0 +1,202 @@
+            ],
+            'PasswordResetRequest new password is at least 8 chars long' => [
+                PasswordResetRequest::fromArray(['user_id' => '1234', 'new_password' => '1234567']),
+                ['new_password' => 'This value is too short. It should have 8 characters or more.'],
+            ],
+
+            'UserRegistrationRequest is valid'                          => [
+                UserRegistrationRequest::fromArray(['email' => 'foo@bar.com', 'password' => '12345678']),
+                [],
+            ],
+            'UserRegistrationRequest values are not blank'              => [
+                UserRegistrationRequest::fromArray(['email' => '', 'password' => '']),
+                [
+                    'email'    => 'This value should not be blank.',
+                    'password' => 'This value should not be blank.',
+                ],
+            ],
+            'UserRegistrationRequest password is at least 8 chars long' => [
+                UserRegistrationRequest::fromArray(['email' => 'foo@bar.com', 'password' => '1234567']),
+                ['password' => 'This value is too short. It should have 8 characters or more.'],
+            ],
+            'UserRegistrationRequest email is valid email address'      => [
+                UserRegistrationRequest::fromArray(['email' => 'some nonsense', 'password' => '12345678']),
+                ['email' => 'This value is not a valid email address.'],
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider providerValidatesInteractorRequests
+     */
+    public function test_it_validates_interactor_requests(AbstractRequest $request, array $expect): void
+    {
+        $subject = $this->newSubject();
+        $this->assertSame($expect, $subject->validate($request));
+    }
+
+    public static function providerValidationRulesInheritedByChildClass(): array
+    {
+        return [
+            [
+                ['email' => 'foo@bar.com', 'password' => '12345678', 'name' => 'Foo'],
+                [],
+            ],
+            [
+                ['email' => '', 'password' => '', 'name' => ''],
+                [
+                    'name'     => 'This value should not be blank.',
+                    'email'    => 'This value should not be blank.',
+                    'password' => 'This value should not be blank.',
+                ],
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider providerValidationRulesInheritedByChildClass
+     */
+    public function test_validation_rules_are_inherited_by_child_class(array $values, array $expect): void
+    {
+        $subject = $this->newSubject();
+        $extended  = new class extends UserRegistrationRequest {
+            #[Assert\NotBlank]
+            protected string $name;
+
+            public function __construct(){
+            }
+        };
+
+        $this->assertSame($expect, $subject->validate($extended::fromArray($values)));
```

Comment:
I had no idea you could do that with an anonymous class, that is super helpful to know